### PR TITLE
fix(backend): tempo cloud connection string

### PR DESCRIPTION
**Pull request by xoscar**
This PR fixes a confusing way of dealing with the tempo cloud http connection, as pasting the given endpoint from the webapp uses `:443` instead of `https`

## Changes

- Defaults `:443` port to `https`

## Fixes

- https://github.com/kubeshop/tracetest/issues/3507

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/tracedb/datasource/http.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ type HttpClient struct {
 
 func NewHttpClient(name string, config *datastore.HttpClientConfig, callback HttpCallback) DataSource {
 	endpoint, _ := urlx.Parse(config.Url)
+	if endpoint.Port() == "443" {
+		endpoint.Scheme = "https"
+	}
```
Comment (xoscar) [inline server/tracedb/datasource/http.go:32]: With a url like this:

`tempo-us-central1.grafana.net:443`

The parsing method returns

`http://tempo-us-central1.grafana.net:443`

Which breaks the `getTraceById` logic.

File: server/tracedb/tempodb.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ import (
 )
 
 func tempoDefaultPorts() []string {
-	return []string{"9095", ""}
+	return []string{"9095", "443", "1"}
```
Comment (xoscar) [inline server/tracedb/tempodb.go:26]: 443 and 1 (using https://tempo-us-central1.grafana.net) should be also valid ports

**Review by jorgeepc — APPROVED** [review]

File: server/tracedb/tempodb.go (hunk 2 above)
Comment (schoren) [reply server/tracedb/tempodb.go:26]: I've never seen port 1 being used. Is there any docs about this port?

Comment (xoscar) [reply server/tracedb/tempodb.go:26]: For some reason the url parser uses `1` as port when using something like `https://tempo-us-central1.grafana.net/`

Comment (xoscar) [reply server/tracedb/tempodb.go:26]: Screen Shot 2024-01-10 at 1 57 31 p m

Comment (schoren) [reply server/tracedb/tempodb.go:26]: That looks like a bug in the URL parsing library. Do you know why we're using an URL parser other than the go's stdlib one?

Comment (xoscar) [reply server/tracedb/tempodb.go:26]: Yeah the standard makes it worse in some cases, here's a table
Screen Shot 2024-01-10 at 2 00 28 p m

Comment (schoren) [reply server/tracedb/tempodb.go:26]: I see. That makes sense. What I would do is handle the case of port "1" and replace it by a correct default port, for example:

```
if port == 1 {
  if schema == "http" {
    port = 80
  }
  if schema == "https" {
    port = 43
  }
}
```

**Comment by adnanrahic** [comment]
This is awesome. When will this be ready for me to try it?

**Pull request merged**